### ISSUE TITLE
Doc: improve documentation on how to use BackgroundJobs

### ISSUE THREAD

**Issue by fgheysels** [arcus-azure/arcus.backgroundjobs#151]
I was going through the documentation regarding the backgroundjob that can be used to detect client-secrets that are about to expire, and it is actually not very clear to me on how I should use it.

The documentation describes:

![image](https://user-images.githubusercontent.com/3605786/176136552-5c59fd61-2691-4c5a-bb74-eda4f4f7ea1e.png)

What exactly does this mean ?  Does this mean that I can use this background-job from within an API for instance ?  Can we elaborate in the documentation a little bit more about this ?
Actually, I personally wouldn't like to do it this way.  Wouldn't it be preferred to create a dedicated component for this ?  Might be a console app which is running in a container or an Azure Function.

Now that I'm typing this, I think we can extend this for all kind of Background Jobs.
I think it would be a good idea to create a page which informs users with best practices on how to implement background jobs, and maybe provide a skeleton that they can use to implement a component that is used to host background jobs.

**Comment by stijnmoreels** [comment]
> What exactly does this mean ? Does this mean that I can use this background-job from within an API for instance ? 

Yes, you could. All background jobs are written as hosted services, so it could be part of a regular job of you application.

> and maybe provide a skeleton that they can use to implement a component that is used to host background jobs

We have already some project templates that use these background jobs, that seems like a good place to put these things, no?

**Comment by stijnmoreels** [comment]
Maybe some user guides can help here, too.

**Comment by stijnmoreels** [comment]
Something I'm thinking about: several of our background jobs are about integrating with other components: auto restart message pump, invalidate secret store... if they are run in another app, they won't have access to the message pump or secret store code.
So, they have to be in the same app for it to work.

**Comment by stijnmoreels** [comment]
Moving this to v1.1.0 to prioritize .NET 8 support.
